modules/moordyn/src/MoorDyn.f90

```diff
@@ -1763,20 +1882,20 @@ SUBROUTINE MD_Init(InitInp, u, p, x, xd, z, other, y, m, DTcoupling, InitOut, Er
             J = J + 1
          
             rRef = m%BodyList(m%CpldBodyIs(l,iTurb))%r6              ! for now set reference position as per input file <<< 
-            !OrMatRef = 
             
             CALL MeshPositionNode(u%CoupledKinematics(iTurb), J, rRef(1:3), ErrStat2, ErrMsg2) ! defaults to identity orientation matrix
-            !TODO: >>> should also maybe set reference orientation (which might make part of a couple lines down redundant) <<<
-            
-            ! calculate initial point relative position, adjusted due to initial platform translations
-            u%CoupledKinematics(iTurb)%TranslationDisp(:,J) = InitInp%PtfmInit(1:3,iTurb) - rRef(1:3)
-            
-            OrMat2 = MATMUL(OrMat, TRANSPOSE( EulerConstruct( rRef(4:6))))  ! combine the Body's relative orientation with the turbine's initial orientation
-            u%CoupledKinematics(iTurb)%Orientation(:,:,J) = OrMat2          ! set the result as the current orientation of the body <<<
 
             ! set absolute initial positions in MoorDyn
-            m%BodyList(m%CpldBodyIs(l,iTurb))%r6(1:3) = u%CoupledKinematics(iTurb)%Position(:,J) + u%CoupledKinematics(iTurb)%TranslationDisp(:,J) + p%TurbineRefPos(:,iTurb)
-            m%BodyList(m%CpldBodyIs(l,iTurb))%r6(4:6) = EulerExtract(MATMUL(OrMat, TRANSPOSE( EulerConstruct( rRef(4:6)))))     ! apply rotation from PtfmInit onto input file's body orientation to get its true initial orientation
+            IF (p%Standalone /= 1) THEN 
+               !TODO: >>> should also maybe set reference orientation (which might make part of a couple lines down redundant) <<<
+               OrMat2 = MATMUL(OrMat, ( EulerConstruct( rRef(4:6))))  ! combine the Body's relative orientation with the turbine's initial orientation
+               u%CoupledKinematics(iTurb)%Orientation(:,:,J) = OrMat2          ! set the result as the current orientation of the body <<<
+
+               ! calculate initial point relative position, adjusted due to initial platform translations
+               u%CoupledKinematics(iTurb)%TranslationDisp(:,J) = InitInp%PtfmInit(1:3,iTurb) - rRef(1:3)
+               m%BodyList(m%CpldBodyIs(l,iTurb))%r6(1:3) = u%CoupledKinematics(iTurb)%Position(:,J) + u%CoupledKinematics(iTurb)%TranslationDisp(:,J) + p%TurbineRefPos(:,iTurb)
```

Comment:
Are these variables initialized elsewhere when `p%Standalone==1`? Or did the driver not initialize some `InitInp` values that are used here? It seems unusual to me that this would be set differently based on how the module is called.

Reply:
They are initialized regardless of the standalone flag. I put them in the if here because the standalone flag is supposed to keep the object positions and orientation in the same location was where they are defined in the input file, and somewhere in these adjustments they were causing different reference frames to be superimposed and messing up the locations of objects defined in the input file. Would it be better practice to leave the MD file as is and just change the driver?

Reply:
Thanks for the info. It sounds like the `InitInp` values may not all be set in the driver like they are when coupled with OpenFAST. I would recommend changing the driver so that this code works the same way regardless of how it is coupled.

Reply:
I looked at things a bit more and I don't think that it's possible to have the standalone feature reside solely in the driver unless the user also defines the coupled objects location in the drive file. With the coupling equations in MD right now, the only way to have the final initialized positions of a coupled object match the MD input file is for the driver to know those positions. Orientations are fine because the driver could just pass the identity matrix as OrMat, but translational positions are determined by a position difference, so zeroing out terms in the driver would lead to negative positions.

https://github.com/RyanDavies19/MoorDynF_ryan/blob/7bdeeb35eb588aaee40c60e9dd08190f8b0d2653/modules/moordyn/src/MoorDyn.f90#L1891

Reply:
We could add a logic statement that makes the positions negative if farm < 1 which should resolve that, but to me that seems more complex than leaving things as is.

Reply:
If I remember correctly, MD has an unusual initialization of positions that was in part dictated by the initial relaxation of the mooring lines.  I vaguely remember exploring how to refactor that code and in the end concluded it was simpler to leave it as it is.  Given that, I can see how this `standalone` flag would be necessary.  I'm ok with leaving this in the module instead of moving it to the driver.